src/module.c

```diff
@@ -10846,10 +10998,407 @@ sds genModulesInfoString(sds info) {
+    int return_code = config->set_fn.set_numeric(config->name, val, config->privdata, err);
+    return return_code == REDISMODULE_OK ? 1 : 0;
+}
+
+/* This is a series of getAndParse functions for each type that, for the most part, emulate the ones
+ * in config.c. The difference here is that we use the get callbacks specified by the module
+ * to get the value. */
+int getModuleBoolConfig(ModuleConfig *module_config) {
+    return module_config->get_fn.get_bool(module_config->name, module_config->privdata);
+}
+
+sds getModuleStringConfig(ModuleConfig *module_config) {
+    RedisModuleString *val = module_config->get_fn.get_string(module_config->name, module_config->privdata);
+    if (val)
+        return sdsdup(val->ptr);
+    return NULL;
+}
+
+int getModuleEnumConfig(ModuleConfig *module_config) {
+    return module_config->get_fn.get_enum(module_config->name, module_config->privdata);
+}
+
+long long getModuleNumericConfig(ModuleConfig *module_config) {
+    return module_config->get_fn.get_numeric(module_config->name, module_config->privdata);
+}
+/*
+This function takes a module and a list of configs stored as sds NAME VALUE pairs.
+ * It attempts to call set on each of these configs. */
+int loadModuleConfigs(RedisModule *module) {
+    if (!server.module_configs_queue || !listLength(server.module_configs_queue)) return REDISMODULE_OK;
+    listIter li;
+    listNode *ln;
+    int count;
+    sds *config_pair;
+    const char *err = NULL;
+
+    listRewind(server.module_configs_queue, &li);
+    while ((ln = listNext(&li))) {
+        sds module_config = ln->value;
+        if (!(strcasestr(module_config, module->name) == module_config)) {
+            continue;
+        }
+        config_pair = sdssplitargs(module_config, &count);
+        if (count != 2) {
+            serverLog(LL_WARNING, "Couldn't load configuration %s", config_pair[0]);
+            sdsfreesplitres(config_pair, count);
+            listDelNode(server.module_configs_queue, ln);
+            while ((ln = listNext(&li))) listDelNode(server.module_configs_queue, ln);
+            return REDISMODULE_ERR;
+        }
+        if (!performInterfaceSet(getConfigFromName(config_pair[0]), config_pair[1], &err)) {
+            serverLog(LL_WARNING, "Issue during loading of configuration %s : %s", config_pair[0], err);
+            sdsfreesplitres(config_pair, count);
+            listDelNode(server.module_configs_queue, ln);
+            while ((ln = listNext(&li))) listDelNode(server.module_configs_queue, ln);
+            return REDISMODULE_ERR;
+        }
+        sdsfreesplitres(config_pair, count);
+        listDelNode(server.module_configs_queue, ln);
+    }
+    return REDISMODULE_OK;
+}
+
+/* Add a moduleConfigTuple to the list if the apply and privdata do not match one already in it. */
+void addModuleApply(list *module_configs_tuples, void *module_config_val) {
+    ModuleConfig *module_config = module_config_val;
```

Comment:
ModuleConfig is in server.h
```suggestion
void addModuleApply(list *module_configs_tuples, ModuleConfig *module_config) {
```